text/stateful-resources-one-pager.md

```diff
@@ -0,0 +1,258 @@
+# 1Pager | Modeling Stateful Resources
+
+This document is a one pager meant to surface the idea of explicitly modeling
+stateful resources in the core framework. Its goal is to conduct a preliminary
+discussion and gather feedback before it potentially progresses into a full blown RFC.
+
+## Customer pain
+
+Customers are currently exposed to unintentional data loss when stateful resources
+are designated for either removal or replacement by CloudFormation. Data, similarly
+to security postures, is an area in which even a single rare mistake can
+cause catastrophic outages.
+
+The CDK can and should help reduce the risk of such failures. With respect to
+security, the CDK currently defaults to blocking deployments that contain
+changes in security postures, requiring a user confirmation:
+
+```console
+This deployment will make potentially sensitive changes according to your current security approval level (--require-approval broadening).
+Please confirm you intend to make the following modifications:
+...
+...
+Do you wish to deploy these changes (y/n)?
+```
+
+However, no such mechanism exists for changes that might result in data loss,
+i.e removal or replacement of stateful resources, such as `S3` buckets, `DynamoDB` tables, etc...
+
+## Failure scenarios
+
+To understand how susceptible customers are to this, we outline a few
+scenarios where such data loss can occur.
+
+### Stateful resource without `DeletionPolicy/UpdateReplacePolicy`
+
+By default, CloudFormation will **delete** resources that are removed from the stack,
+or when a property that requires replacement is changed.
+
+> - [UpdateReplacePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatereplacepolicy.html)
+> - [DeletionPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html)
+
+To retain stateful resources, authors must remember to configure those policies
+with the `Retain` value. Having to remember this makes it also easy to forget.
+This means that stateful resources might be shipped with the incorrect policy.
```

Comment:
Most of this has been implemented in https://github.com/aws/aws-cdk/pull/12920

Reply:
Taking a loog.